Applications/Spire/Source/Ui/FlatButton.cpp

```diff
@@ -12,118 +12,134 @@ FlatButton::FlatButton(QWidget* parent)
-  m_default_style.m_text_color = Qt::black;
-  m_hover_style.m_background_color = Qt::white;
-  m_hover_style.m_border_color = Qt::transparent;
-  m_hover_style.m_text_color = Qt::black;
-  m_focus_style.m_background_color = Qt::white;
-  m_focus_style.m_border_color = Qt::transparent;
-  m_focus_style.m_text_color = Qt::black;
-  m_disabled_style.m_background_color = Qt::transparent;
-  m_disabled_style.m_border_color = Qt::white;
-  m_disabled_style.m_text_color = Qt::white;
-  set_style(m_default_style);
-  set_hover_style(m_hover_style);
-  set_focus_style(m_focus_style);
-  set_disabled_style(m_disabled_style);
+  auto button_font = QFont("Roboto");
+  button_font.setPixelSize(scale_height(12));
+  button_font.setWeight(QFont::Normal);
+  button_font.setLetterSpacing(QFont::AbsoluteSpacing, 0);
+  m_label->setFont(button_font);
+  set_style({QColor("#EBEBEB"), QColor("#EBEBEB"), QColor("#000000")},
+    {QColor("#4B23A0"), QColor("#4B23A0"), QColor("#FFFFFF")},
+    {QColor("#EBEBEB"), QColor("#4B23A0"), QColor("#000000")},
+    {QColor("#EBEBEB"), QColor("#EBEBEB"), QColor("#A0A0A0")});
 }
 
 void FlatButton::set_label(const QString& label) {
   m_label->setText(label);
 }
 
-const FlatButton::Style& FlatButton::get_style() const {
+void FlatButton::set_style(const Style& default_style, const Style& hover_style,
+    const Style& focus_style, const Style& disabled_style) {
+  m_default_style = default_style;
+  m_hover_style = hover_style;
+  m_focus_style = focus_style;
+  m_disabled_style = disabled_style;
+  setStyleSheet(QString(R"(
+    #flat_button {
+      background-color: %1;
+      border: %2px solid %3;
+      margin: 0px;
+    }
+    #flat_button:hover {
+      background-color: %4;
+      border-color: %5;
+    }
+    #flat_button:focus:hover {
+      background-color: %4;
+      border-color: %5;
+    }
+    #flat_button:focus {
+      background-color: %6;
+      border-color: %7;
+    }
+    #flat_button:disabled {
+      background-color: %8;
+      border-color: %9;
+    })")
+    .arg(m_default_style.m_background_color.name(QColor::HexArgb))
+    .arg(scale_width(1))
+    .arg(m_default_style.m_border_color.name(QColor::HexArgb))
+    .arg(m_hover_style.m_background_color.name(QColor::HexArgb))
+    .arg(m_hover_style.m_border_color.name(QColor::HexArgb))
+    .arg(m_focus_style.m_background_color.name(QColor::HexArgb))
+    .arg(m_focus_style.m_border_color.name(QColor::HexArgb))
+    .arg(m_disabled_style.m_background_color.name(QColor::HexArgb))
+    .arg(m_disabled_style.m_border_color.name(QColor::HexArgb)));
+  set_label_stylesheet(false);
+}
+
+const FlatButton::Style& FlatButton::get_default_style() const {
   return m_default_style;
 }
 
-void FlatButton::set_style(const Style& default_style) {
-  m_default_style = default_style;
-  on_style_updated();
+void FlatButton::set_default_style(const Style& default_style) {
+  set_style(default_style, m_hover_style, m_focus_style, m_disabled_style);
```

Comment:
The purpose of creating the set_style function with all styles passing in is to set the style up once to avoid calling set_default_style(...), set_hover_style(...), etc. Multiple calls of these functions can Implicitly call setStyleSheet multiple times. In practice, the setting generally only occurs when the button is initialized and rarely changes in runtime. I wanted to add them to the constructor, but I gave up because it would make the constructor take too many parameters. However, these set_x_style functions can make the use of the buttons more flexible, and I will add a private function according to your suggestion.

Reply:
Yes, if set_x_style(...) is called for every style, then there will be a stylesheet update for each call.

My suggestion is purely about maintenance, because it removes the need to explicitly list every parameter in every set_x_style(...) function. For example, the set_default_style(...) function would look like this:

```
void FlatButton::set_default_style(const QStyle& default_style) {
    m_default_style = default_style;
    update_style();
}

// ... 3 other style setters ...

void FlatButton::update_style() {
    set_style(m_default_style, m_hovered_style, m_focused_style, m_disabled_style);
}
```

The advantage now is that if we ever had to modify this class every setter leads to the same outcome, so it removes the chance of making a mistake when updating the parameters in the 4 calls to set_style(...).